src/Microsoft.DotNet.Build.VstsBuildsApi/VstsReleaseHttpClient.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Build.VstsBuildsApi
+{
+    internal class VstsReleaseHttpClient : IVstsDefinitionHttpClient
+    {
+        private static readonly string[] s_InstanceIdentifiableField = {
+            "_links",
+            "createdBy",
+            "createdOn",
+            "deployStep",
+            "modifiedBy",
+            "modifiedOn",
+            "owner",
+            "id",
+            "lastRelease",
+            "revision",
+        };
+
+        private const string ApiVersion = "3.1-preview.3";
+
+        private JsonHttpClient _jsonClient;
+
+        public VstsReleaseHttpClient(JObject definition, string credentials)
+        {
+            Uri uri = new Uri(GetVstsReleaseApiUrl(definition));
+
+            _jsonClient = new JsonHttpClient(
+                new Uri(string.Format("{0}://{1}/", uri.Scheme, uri.Authority)),
+                credentials);
+        }
+
+        public async Task<JObject> VstsUpdateDefinitionAsync(JObject definition)
+        {
+            string requestUri = GetRequestUri(definition, "definitions");
+            return await _jsonClient.PutAsync(requestUri, definition);
+        }
+
+        public async Task<JObject> VstsCreateDefinitionAsync(JObject definition)
+        {
+            string requestUri = GetRequestUri(definition, "definitions");
+            return await _jsonClient.PostAsync(requestUri, definition);
+        }
+
+        public async Task<JObject> VstsRetrieveDefinitionByIdAsync(JObject definition)
+        {
+            string requestUri = GetRequestUri(definition, $"definitions/{definition["id"]}");
+            return await _jsonClient.GetAsync(requestUri);
+        }
+
+        public async Task<IReadOnlyList<JObject>> VstsRetrieveDefinitionsListByNameAndPathAsync(JObject definition)
```

Comment:
Ah, dang, I guess my personal account's VSTS instance is a bit ahead then. I'll test Release creation very soon on devdiv.visualstudio.com to make sure that isn't different too.